_posts/2021-02-09-non-key-joins-in-kstreams.adoc

```diff
@@ -0,0 +1,288 @@
+Running Quarkus applications in native code via GraalVM has a very fast start-up time, the applications use significantly less memory when compiled to native code. This allows you to start as many instances of a Kafka Streams pipeline in parallel in a very memory-efficient way.
+If you want to run this application in `native` mode, set the `QUARKUS_MODE` as `native` and run:
+
+----
+mvn clean package -f aggregator/pom.xml -Pnative
+----
+Here is a https://quarkus.io/guides/kafka-streams#running-natively[detailed guide] for you.
+
+== More Insights on Kafka Streams Extension
+This extension can help you address some of the common requirements when building microservices.
+For running your Kafka Streams application in production, you can also add health checks and metrics for the data pipeline.
+
+https://quarkus.io/guides/microprofile-metrics[Metrics] allow applications to gather various metrics and statistics that provide insights into what is happening inside the application.
+Using the MicroProfile Metrics API, these metrics can be exposed via HTTP using JSON format or the OpenMetrics format.
+From there they can be scraped by tools such as Prometheus and stored for analysis and visualization.
+Apart from application-specific metrics, you can utilize built-in metrics exposed by various Quarkus extensions.
+
+On the other hand, we have https://quarkus.io/guides/microprofile-health[MicroProfile Health], which provides information about the liveness of the application i.e. states whether your application is running or not and whether your application is able to process requests.
+
+Along with Kafka Streams https://quarkus.io/guides/kafka-streams#interactive-queries[Interactive Queries], you can directly query the underlying state store of the pipeline for the value associated with a given key.
+By exposing a simple REST endpoint that queries the state store, the latest aggregation result can be retrieved without having to subscribe to any Kafka topic.
+
+== Summary
+The Quarkus extension for Kafka Streams comes with everything needed to run stream processing pipelines on the JVM as well as in Native mode, along with additional bonuses of performing health checks, metrics and interactive queries.
+
+In this article we have discussed stream processing topology of foreign key joins in Kafka Streams and how to use the Quarkus Kafka Streams extension for running and building your application in JVM mode.
+
+You can find the complete https://github.com/debezium/debezium-examples/tree/master/kstreams-fk-join[source code] of the implementation in the Debezium examples repo.
+If you got any questions or feedback, please let us know in the comments below.
+We're looking forward to your suggestions!
+
+_Thanks a lot https://twitter.com/gunnarmorling/[Gunnar Morling] for your feedback while working on this post!_
```

Comment:
Thanks for the shout-out! But I'd gently suggest to remove it, it's your work and it stands fine by itself.